### ISSUE TITLE
Add way to query a Table

### ISSUE THREAD

**Issue by turicas** [turicas/rows#148]
Sometimes is really handy to just have a `rows.Table` filtered (like we would do using Django ORM with the `.objects.filter` method).

I don't know if the best option for `rows` API is to add a `filter`/`query` method into `rows.Table` (will return a new `rows.Table` object) or just create a helper function. In my scripts I'm currently creating helper functions so I can feel better how the API should be.

The implementation would be something like this:

```python
def filter_row(row, kwargs):
    for key, value in kwargs.items():
        if getattr(row, key) != value:
            break
    else:
        return True
    return False


def query_table(table, *args, **kwargs):
    new_table = rows.Table(fields=table.fields)
    # TODO: should also add _meta and other Table attributes
    new_table.extend(row._asdict() for row in table if filter_row(row, kwargs))
    return new_table
```

**Comment by humrochagf** [comment]
I think the use of the filter style implementation would be interesting to do things like lazy evaluation and to provide a default extensible api to people implent your own queryset

**Comment by turicas** [comment]
@humrochagf, good point! I was thinking in returning a new `rows.Table` with filtered data, but I can simply return a queryset (which can be filtered again, and again, and again). I'd like to have a better solution for this into version 0.3.0 (the next one), do you have any ideas on the API?
